src/tests/classic/run.rs

```diff
@@ -256,3 +279,206 @@ fn test_forms_of_destructuring_allowed_by_classic_1() {
         "(i 2 (q . 2) (q . 3))"
     );
 }
+
+#[test]
+fn test_num_encoding_just_less_than_5_bytes() {
+    let res = do_basic_run(&vec!["run".to_string(), "4281419728".to_string()])
+        .trim()
+        .to_string();
+    assert_eq!(res, "0x00ff3147d0");
+}
+
+#[test]
+fn test_divmod() {
+    let res = do_basic_run(&vec![
+        "run".to_string(),
+        "(/ 78962960182680 4281419728)".to_string(),
+    ])
+    .trim()
+    .to_string();
+    assert_eq!(res, "18443");
+}
+
+#[cfg(test)]
+pub struct RandomClvmNumber {
+    pub intended_value: Number,
+}
+
+#[cfg(test)]
+pub fn random_clvm_number<R: Rng + ?Sized>(rng: &mut R) -> RandomClvmNumber {
+    // Make a number by creating some random atom bytes.
+    // Set high bit randomly.
+    let natoms = rng.gen_range(0..=NUM_GEN_ATOMS);
+    let mut result_bytes = Vec::new();
+    for _ in 0..=natoms {
+        let mut new_bytes = sexp::random_atom_name(rng, 3)
+            .iter()
+            .map(|x| {
+                if rng.gen() {
+                    // The possibility of negative values.
+                    x | 0x80
+                } else {
+                    *x
+                }
+            })
+            .collect();
+        result_bytes.append(&mut new_bytes);
+    }
+    let num = number_from_u8(&result_bytes);
+
+    RandomClvmNumber {
+        intended_value: num,
+    }
+}
+
+#[cfg(test)]
+impl Distribution<RandomClvmNumber> for Standard {
+    fn sample<R: Rng + ?Sized>(&self, rng: &mut R) -> RandomClvmNumber {
+        random_clvm_number(rng)
+    }
+}
+
+// Finally add property based testing in here.
+#[test]
+fn test_encoding_properties() {
+    let mut rng = ChaChaRng::from_entropy();
+    for _ in 1..=200 {
+        let number_spec: RandomClvmNumber = rng.gen();
+
+        // We'll have it compile a constant value.
+        // The representation of the number will come out most likely
+        // as a hex constant.
+        let serialized_through_run = do_basic_run(&vec![
+            "run".to_string(),
+            format!("(q . {})", number_spec.intended_value),
+        ])
+        .trim()
+        .to_string();
```

Comment:
this test isn't quite as good as a unit test. it's not targeting edge cases (e.g. byte-boundary values). It also "launders" possible encodings with redundant leading zeroes or `\xff` bytes, since you just compare the value after it's interpreted as an integer again.

If it fails, it's also hard to reproduce since the random sequence is different for every run.